<launch>
	<arg name="uavname" default="fpv" />

	<node pkg="mockamap" type="mockamap_node" name="mockamap_node" output="screen">
  <param name="seed" type="int" value="1024"/>
  <param name="update_freq" type="double" value="1.0"/>
  <param name="resolution" type="double" value="0.25"/>
  <param name="x_length" type="int" value="50"/>
  <param name="y_length" type="int" value="50"/>
  <param name="z_length" type="int" value="5"/>
  <param name="type" type="int" value="1"/>
  <param name="complexity"    type="double" value="0.025"/>
  <param name="fill"          type="double" value="0.3"/>
  <param name="fractal"       type="int"    value="1"/>
  <param name="attenuation"   type="double" value="0.1"/>
  <remap from="/mock_map" to="/voxel_map"/>
  </node>

	<node pkg="px4_controller" type="px4_control_node" name="px4_controller" output="screen">
		<remap from="~state" to="/mavros/state" />
		<remap from="~odom" to="/mavros/local_position/odom" />
		<remap from="~cmd" to="/position_command" />
		<remap from="~imu" to="/mavros/imu/data" />
		<rosparam command="load" file="$(find px4_controller)/config/ctrl_param_$(arg uavname).yaml" />
	</node>
	
	<!-- MAVROS posix SITL environment launch script -->
	<!-- launches MAVROS, PX4 SITL, Gazebo environment, and spawns vehicle -->
	<!-- vehicle pose -->
	<arg name="x" default="0.0"/>
	<arg name="y" default="0.0"/>
	<arg name="z" default="0.4"/>
	<arg name="R" default="0"/>
	<arg name="P" default="0"/>
	<arg name="Y" default="0"/>
	<!-- gazebo configs -->
	<arg name="gui" default="true"/>
	<arg name="debug" default="false"/>
	<arg name="verbose" default="false"/>
	<arg name="paused" default="false"/>
	<arg name="respawn_gazebo" default="false"/>
	<!-- vehicle model and world -->
	<arg name="est" default="ekf2"/>
	<arg name="vehicle" default="iris"/>
	<!-- <arg name="world" default="$(find px4_controller)/worlds/hill_cable_tree.world"/> -->
	<!-- <arg name="sdf" default="$(find px4_controller)/models/$(arg model)/$(arg model).sdf"/> -->
	<!-- MAVROS configs -->
	<arg name="fcu_url" default="udp://:14540@localhost:14557"/>
	<arg name="respawn_mavros" default="false"/>
	<!-- PX4 configs -->
	<arg name="interactive" default="true"/>
	<!-- PX4 SITL and Gazebo -->
	<include file="$(find px4)/launch/posix_sitl.launch">
			<arg name="x" value="$(arg x)"/>
			<arg name="y" value="$(arg y)"/>
			<arg name="z" value="$(arg z)"/>
			<arg name="R" value="$(arg R)"/>
			<arg name="P" value="$(arg P)"/>
			<arg name="Y" value="$(arg Y)"/>
			<!-- <arg name="world" value="$(arg world)"/> -->
			<!-- <arg name="vehicle" value="$(arg vehicle)"/> -->
			<!-- <arg name="sdf" value="$(arg sdf)"/> -->
			<arg name="gui" value="$(arg gui)"/>
			<arg name="interactive" value="$(arg interactive)"/>
			<arg name="debug" value="$(arg debug)"/>
			<arg name="verbose" value="$(arg verbose)"/>
			<arg name="paused" value="$(arg paused)"/>
			<arg name="respawn_gazebo" value="$(arg respawn_gazebo)"/>
	</include>
	<!-- MAVROS -->
	<include file="$(find mavros)/launch/px4.launch">
			<!-- GCS link is provided by SITL -->
			<arg name="gcs_url" value=""/>
			<arg name="fcu_url" value="$(arg fcu_url)"/>
			<arg name="respawn_mavros" value="$(arg respawn_mavros)"/>
	</include>
</launch>
